File: src/main/java/jmpolak/task_notification/core/entity/Task.java
package jmpolak.task_notification.core.entity;
import java.time.LocalDateTime;

import lombok.AllArgsConstructor;
import lombok.Getter;
@Getter
@AllArgsConstructor
public class Task {
    String id; // optional - from db
    String title;
    String note;
    String to;
    boolean attachment;
    boolean completed;
    LocalDateTime notificationDate;
    LocalDateTime createDate;

    public Task(String title, String note, String to, boolean attachment, boolean completed, LocalDateTime notificationDate){
        this.title = title;
        this.note = note;
        this.to = to;
        this.attachment = attachment;
        this.completed = completed;
        this.notificationDate = notificationDate;
        this.createDate = LocalDateTime.now();
    }
}

File: frontend/src/app/core/api/task/task.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

interface ResponseTaskDto {
  id: string; // optional - from db
  title: string;
  note: string;
  to: string;
  attachment: boolean;
  completed: boolean;
  notificationDate: string;
  createDate: string;
}
@Injectable({
  providedIn: 'root',
})
export class Task {
  private readonly url = 'api/task';
  constructor(private http: HttpClient) {}

  getTask(date: string) {
    return this.http.get<ResponseTaskDto[]>(`${this.url}/by-date?date=${date}`);
  }
}
